src/sdk/namespace_fs.js

```diff
@@ -863,6 +908,10 @@ class NamespaceFS {
                 await target_file.setxattr(fs_account_config, fs_xattr);
             }
             if (config.NSFS_TRIGGER_FSYNC) await target_file.fsync(fs_account_config);
+            dbg.log1('NamespaceFS.upload_stream:', open_mode, file_path, upload_path);
+            if (open_mode === 'wt') {
+                await target_file.linkfileat(fs_account_config, file_path);
```

Comment:
almost forgot - we need to make path dirs for file_path, right before we call linkfileat:
```js
if (open_mode === 'wt') {
    await this._make_path_dirs(file_path, fs_account_config);
    await target_file.linkfileat(fs_account_config, file_path);
}
```

Reply:
the directory must exist before the open, the link will not move it to another directory, so if the directory did not exist we should create it before open in _upload_stream()

Reply:
Let me try to understand this better - 
- What are the restrictions on path for `open(path, 'wt')`? 
- Does it need to be a directory or a file path?
- If the dir path of it has to exist, then we will have to call `_make_path_dirs` there too, which we currently don't have in the 'wt' flow at all, so I think we need to add. 
- Currently I see that we send the `bucket_path` (which is the root dir of the bucket) as the path to open tmp file. We only use the target `file_path` (which is the object key) for `linkfileat()`, so I am confused with your last comment that `link` will not move it to another directory.
- Do we have to provide the `file_path` to open instead of the `bucket_path`?
- We will need to call `_make_path_dirs` again before `linkfileat` to ensure that the dirs in `file_path` still exist after all the upload time has passed - we had a bug like this already in the rename flow that the dirs were deleted by the time we finished writing to the upload_path, and so we had to create those right before renaming to the target.

Reply:
The path for open is a directory path where the temp file will be created. We want to use the parent directory so we can inherent attributes from it. I don't think you will be able to delete the directory while a file is open in it, this different from creating a file in work directory and than rename it to a directory that might not exist anymore. I will try to fix it.

Reply:
so you mean that linkat will fail if the dir path differs from open - like this ? 
```
fd = open("dir1/", "wt");
linkat(fd, "", AT_FDCWD, "dir1/file", AT_EMPTY_PATH); // OK
linkat(fd, "", AT_FDCWD, "dir2/file", AT_EMPTY_PATH); // ERROR ?
```

Reply:
or should we provide `"dir1/file"` to open as well?

Reply:
temp open expects a directory path, in this case dir1 NOT dir1/file
the link will work with a different path, like dir2/file2 but it is better to create the temp file in the target directory so it can inherent the directory attributes.   
I will work on a fix to open the right director.

Reply:
I think we still need to mkdirs here too because what if that dir was deleted/moved during our upload?

Reply:
the directory can not be deleted, if it is moved, which is very unlikely, the request will fail.

Reply:
OK, I wonder - since linkat is actually taking a target dirfd, which we always send as AT_FDCWD, perhaps we can open that dir and keep the dirfd while uploading, and at the end here, we will send dirfd+filename which will make it precisely refer to the opened target dir. WDYT?